demos/noise_suppression_demo/python/noise_suppression.py

```diff
@@ -0,0 +1,131 @@
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    log.info("Initializing Inference Engine")
+    ie = IECore()
+    version = ie.get_versions(args.device)[args.device]
+    version_str = "{}.{}.{}".format(version.major, version.minor, version.build_number)
+    log.info("Plugin version is {}".format(version_str))
+
+    # read IR
+    model_xml = args.model
+    model_bin = model_xml.with_suffix(".bin")
+    log.info("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+
+    # check input and output names
+    input_shapes = {k: v.input_data.shape for k, v in ie_encoder.input_info.items()}
+    input_names = list(ie_encoder.input_info.keys())
+    output_names = list(ie_encoder.outputs.keys())
+
+    assert "input" in input_names, "'input' is not presented in model"
+    assert "output" in output_names, "'output' is not presented in model"
+    state_inp_names = [n for n in input_names if "state" in n]
+    state_param_num = sum(np.prod(input_shapes[n]) for n in state_inp_names)
+    log.info("state_param_num = {} ({:.1f}Mb)".format(state_param_num, state_param_num*4e-6))
+
+    # load model to the device
+    log.info("Loading model to the {}".format(args.device))
+    ie_encoder_exec = ie.load_network(network=ie_encoder, device_name=args.device)
+
+    sample_inp, freq = soundfile.read(args.input, start=0, stop=None, dtype='float32')
+    assert freq == 16000, "Only 16kHz sampling rate is supported for input audio"
+    if sample_inp.ndim == 2:
+        sample_inp = sample_inp.mean(0)
+
+    input_size = input_shapes["input"][1]
+    res = None
+
+    samples_out = []
+    samples_times = []
+    while sample_inp is not None and sample_inp.shape[0] > 0:
+        if sample_inp.shape[0] > input_size:
+            input = sample_inp[:input_size]
+            sample_inp = sample_inp[input_size:]
+        else:
+            input = np.pad(sample_inp, ((0, input_size - sample_inp.shape[0]), ), mode='constant')
+            sample_inp = None
+
+        #forms input
+        inputs = {"input": input[None, :]}
+
+        #add states to input
+        for n in state_inp_names:
+            if res:
+                inputs[n] = res[n.replace('inp', 'out')]
+            else:
+                #on the first iteration fill states by zeros
+                inputs[n] = np.zeros(input_shapes[n], dtype=np.float32)
+
+        t0 = time.perf_counter()
+        # Copy inputs manually through InferRequest functionality to speedup
+        infer_request_ptr = ie_encoder_exec.requests[0]
+        for n, data in inputs.items():
+            info_ptr = ie_encoder.input_info[n]
+            blob = Blob(info_ptr.tensor_desc, data)
+            infer_request_ptr.set_blob(n, blob, info_ptr.preprocess_info)
+
+        # infer by IE
+        res = ie_encoder_exec.infer()
```

Comment:
If the workaround with copying is kept, you should do all manipulations including infere() with infer_request. Don't jump between two approaches of infer_request and exec_net.